testsuite/test_student_workflow.py

```diff
@@ -344,3 +344,30 @@ def test_user_see_revealed_score(client, STC, base_db_setup):
     )
     STC.assertNotContains(response, "N/A")
     STC.assertContains(response, "Final score")
+
+
+@pytest.mark.django_db
+def test_user_code_view(client, STC, base_db_setup):
+    FILEPATH = settings.BASE_DIR / 'testsuite' / 'test.zip'
+
+    (_, _, user, course, assignment) = base_db_setup
+
+    client.login(username=user.username, password='pw')
+
+    with open(FILEPATH, 'rb') as file:
+        upload = SimpleUploadedFile(FILEPATH.name,
+                                    file.read(),
+                                    content_type='application/zip')
+        client.post(
+            f'/assignment/{course.pk}/{assignment.pk}/upload/',
+            data={'name': FILEPATH.name, 'file': upload},
+            format='multipart',
+        )
+
+    submission = Submission.objects.all()[0]
+
+    response = client.get(
+        f'/submission/{submission.pk}/test',
+        follow=True,
+    )
+    STC.assertTemplateNotUsed(response, 'interface/code_view.html')
```

Comment:
You could add a check for the response status and if it redirects to the admin page.